julia-mode.el

```diff
@@ -67,6 +67,35 @@ User can still use `abbrev-mode' or `expand-abbrev' to substitute
 unicode for LaTeX even if disabled."
   :type 'boolean)
 
+(defconst julia-mode--latexsubs-partials
+  (let ((table (make-hash-table :test 'equal)))
+    (maphash (lambda (latex subst)
+               (cl-assert (string= (substring latex 0 1) "\\") nil
+                          "LaTeX substitution does not start with \\.")
+               (let ((len (length latex)))
+                 (cl-assert (< 1 len) nil "Trivially short LaTeX subtitution")
+                 (cl-loop for i from 2 to len
+                          do (puthash (substring latex 1 i) t table))))
+             julia-mode-latexsubs)
+    table)
+  "A hash table containing all partial strings from the LaTeX abbreviations in `julia-mode-latexsubs' as keys. Values are always `t', the purpose is to represent a set.")
+
+(defun julia-mode--latexsubs-longest-partial-end (beg)
+  "Starting at `beg' (should be the  \"\\\"), return the end of the longest partial match for LaTeX completion, or `nil' when not applicable."
+  (save-excursion
+    (goto-char beg)
+    (when (and (= (char-after) ?\\) (not (eobp)))
+      (forward-char)
+      (let ((beg (point)))
+        (defun next-char-matches? ()
+          (let* ((end (1+ (point)))
+                 (str (buffer-substring-no-properties beg end))
+                 (valid? (gethash str julia-mode--latexsubs-partials)))
+            valid?))
+        (while (and (not (eobp)) (next-char-matches?))
```

Comment:
Byte-compiler complains that `next-char-matches?` isn't known to be defined, and I'm not entirelys sure why. Could either move it to a top level function that takes `beg` as an argument or just put all the logic from `next-char-matches?` inline here.